src/main/java/com/newrelic/experts/jenkins/EventRecorderImpl.java

```diff
@@ -39,6 +40,8 @@
 
   private static final String CLASS_NAME = EventRecorderImpl.class.getName();
   private static final Logger LOGGER = Logger.getLogger(CLASS_NAME);
+  private final boolean isLoggingTrace = LOGGER.isLoggable(Level.FINE);
+  private final boolean isLoggingDebug = LOGGER.isLoggable(Level.FINEST);
```

Comment:
@dalvizu The `EventRecordImpl` is bound as a singleton object into the DI container.  If we hold onto trace flags like this at the instance level, changes to log levels would not be picked up.  It would always use the log level that was in effect when the instance was initialized.  Please undo these changes and leave the original implementation.